.github/workflows/terraform-docs.yaml

```diff
@@ -2,20 +2,33 @@ name: terraform-docs
 on:
   push:
     branches: [ main ]
+  pull_request:
```

Comment:
don't think we want this.

Reply:
Github workflow unable to push commits to MAIN and the workflow is failing due to our restrictions.
- No commits to main
- at-least one reviewer is required

It's also good to see what's being pushed to MAIN by looking at the PR. Hence i changed this to  on pull request but this will be executed only when the PR raised within the same repo (tfdocs wont be run on PRs raised from fork)

This PR worked with the latest workflow